x/auth/vesting/types/vesting_account.go

```diff
@@ -891,11 +895,13 @@ func (va *ClawbackVestingAccount) clawback(ctx sdk.Context, dest sdk.AccAddress,
 		return nil
 	}
 	addr := va.GetAddress()
+	ak.SetAccount(ctx, va)
 
 	// Now that future vesting events (and associated lockup) are removed,
 	// the balance of the account is unlocked and can be freely transferred.
 	spendable := bk.SpendableCoins(ctx, addr)
 	toXfer := coinsMin(toClawBack, spendable)
```

Comment:
Need to swap these two lines out or just delete them, due to gas exhaustion attacks.

Reply:
The check for spendable coins is meant to be defensive. The thought is that in case of a logic error, we try to get as much as we can (up to the  unvested amount). Without the spendable check, you might not be able to claw back anything. Possible sources of error might be rounding or a logic bug in rewriting the vesting schedule.